File: GanAndroidApp/app/src/main/java/com/xattacker/gan/data/FunctionType.java
package com.xattacker.gan.data;

public enum FunctionType
{
	UNKNOWN_TYPE(-1),
	
	LOGIN(1),
	LOGOUT(2),
	REGISTER_ACCOUNT(3),
	CONNECTION(4),
	SEND_SMS(5),
	RECEIVE_SMS(6),
	GET_IP(7),
	GET_SYSTEM_TIME(8),
	GET_FRIEND_LIST(9),
	ADD_FRIEND(10),
	CHECK_ACCOUNT(11);
	
	
   private int _value;
   
   private FunctionType(int value)
   {
   	_value = value;
   }
   
   public int value()
   {
  	 	return _value;
   }
   
	public static FunctionType parse(int aValue)
	{
		for (FunctionType type : FunctionType.values())
		{
			if (type.value() == aValue)
			{
				return type;
			}
		}

		return UNKNOWN_TYPE;
	}
}

File: GaniOSApp/GanClient/GanClient/Gan/Data/FunctionType.swift
//
//  FunctionType.swift
//  GanClient
//
//  Created by xattacker on 2020/11/23.
//

internal enum FunctionType: Int
{
    case unknown = -1
    
    case login = 1
    case logout = 2
    case register_account = 3
    case get_account_list = 4
    case check_account = 5
    
    case send_sms = 101
    case receive_sms = 102
    case receive_sms_ack = 103
    
    case get_ip = 901
    case get_system_time = 902
}
